Branch-SDK/Branch-SDK/BranchLinkProperties.m

```diff
@@ -0,0 +1,29 @@
+//
+//  BranchLinkProperties.m
+//  Branch-TestBed
+//
+//  Created by Derrick Staten on 10/16/15.
+//  Copyright © 2015 Branch Metrics. All rights reserved.
+//
+
+#import "BranchLinkProperties.h"
+
+@implementation BranchLinkProperties
+
+- (NSDictionary *)controlParams {
+    if (!_controlParams) {
+        _controlParams = [[NSDictionary alloc] init];
+    }
+    return _controlParams;
+}
+
+- (void)addControlParam:(NSString *)controlParam withValue:(NSString *)value {
+    if (!controlParam || ! value) {
```

Comment:
what are you checking with `!controlParam` and `!value`? if you're going to check null ness and emptiness, I recommend you compare agains `[NSNull null]` and then check length of string.